daemon/src/maker_cfd_actor.rs

```diff
@@ -1,216 +1,344 @@
+        insert_order(&order, &mut conn).await.unwrap();
+
+        // 2. Update actor state to current order
+        self.current_order_id.replace(order.id);
+
+        // 3. Notify UI via feed
+        self.order_feed_sender.send(Some(order.clone()))?;
+
+        // 4. Inform connected takers
+        self.takers
+            .as_ref()
+            .expect("actor to be initialized")
+            .do_send_async(maker_inc_connections_actor::BroadcastOrder(Some(order)))
+            .await?;
+        Ok(())
+    }
+
+    async fn handle_start_contract_setup(
+        &mut self,
+        msg: StartContractSetup,
+        ctx: &mut Context<Self>,
+    ) -> anyhow::Result<()> {
+        let StartContractSetup { taker_id, order_id } = msg;
+
+        // Kick-off the CFD protocol
+        let (sk, pk) = crate::keypair::new(&mut rand::thread_rng());
+
+        let mut conn = self.db.acquire().await.unwrap();
+
+        let cfd = load_cfd_by_order_id(order_id, &mut conn).await.unwrap();
+        let margin = cfd.margin().unwrap();
+
+        let maker_params = self.wallet.build_party_params(margin, pk).unwrap();
+
+        let (actor, inbox) = setup_contract_actor::new(
+            {
+                let inbox = self
+                    .takers
+                    .as_ref()
+                    .expect("actor to be configured")
+                    .clone();
+                move |msg| {
+                    inbox.send(maker_inc_connections_actor::OutProtocolMsg { taker_id, msg });
                 }
+            },
+            setup_contract_actor::OwnParams::Maker(maker_params),
+            sk,
+            self.oracle_pk,
+            cfd,
+        );
+
+        self.current_contract_setup = Some(inbox);
+        ctx.notify(CfdSetupCompleted(actor.await));
+        Ok(())
+    }
+
+    async fn handle_new_taker_online(&mut self, msg: NewTakerOnline) -> anyhow::Result<()> {
+        let mut conn = self.db.acquire().await.unwrap();
+
+        let current_order = match self.current_order_id {
+            Some(current_order_id) => Some(load_order_by_id(current_order_id, &mut conn).await?),
+            None => None,
+        };
+
+        self.takers
+            .as_ref()
+            .expect("Actor to be configured")
+            .do_send_async(maker_inc_connections_actor::SendOrder {
+                order: current_order,
+                taker_id: msg.id,
+            })
+            .await?;
+
+        Ok(())
+    }
+
+    async fn handle_inc_protocol_msg(&mut self, msg: IncProtocolMsg) -> anyhow::Result<()> {
+        let inbox = match &self.current_contract_setup {
+            None => panic!("whoops"),
+            Some(inbox) => inbox,
+        };
+        inbox.send(msg.0)?;
+        Ok(())
+    }
+
+    async fn handle_sync_wallet(&mut self) -> anyhow::Result<()> {
+        let wallet_info = self.wallet.sync().unwrap();
+        self.wallet_feed_sender.send(wallet_info).unwrap();
+        Ok(())
```

Comment:
https://doc.rust-lang.org/unstable-book/language-features/try-blocks.html would be nice to have here but we can abuse `async` blocks:

```suggestion
		if let Err(e) = async {
	          let wallet_info = self.wallet.sync()?;
	          self.wallet_feed_sender.send(wallet_info)?;
	          
	          Ok(())
		}.await;
```

Reply:
That should allow us to do stuff directly inside the handler functions without having to wrap them all!

Reply:
I tried using it, and it was complaining about the need of using fully qualified types.... if you don't mind, I'd be happy to add a small PR on top of these changes that cleans the handlers a bit - now I'm trying to merge before new rebase conflicts :)